generic3g/vertical/ModelVerticalGrid.F90

```diff
@@ -65,89 +63,99 @@ module function can_connect_to(this, src, rc)
 
 contains
 
-   function new_ModelVerticalGrid_basic(num_levels, units) result(vgrid)
+   function new_ModelVerticalGrid_basic(standard_name, units, num_levels) result(vgrid)
       type(ModelVerticalGrid) :: vgrid
-      integer, intent(in) :: num_levels
+      character(*), intent(in) :: standard_name
       character(*) , intent(in) :: units
-      !# character(*), intent(in) :: short_name
-      !# character(*), intent(in) :: standard_name
-      !# type(StateRegistry), pointer, intent(in) :: registry
+      integer, intent(in) :: num_levels
 
       call vgrid%set_id()
+      vgrid%standard_name = standard_name
       call vgrid%set_units(units)
       vgrid%num_levels = num_levels
-      !# vgrid%short_name = short_name
-      !# vgrid%standard_name = standard_name
-      !# vgrid%registry => registry
    end function new_ModelVerticalGrid_basic
 
    integer function get_num_levels(this) result(num_levels)
       class(ModelVerticalGrid), intent(in) :: this
       num_levels = this%num_levels
    end function get_num_levels
 
-   subroutine add_variant(this, short_name)
+   subroutine add_short_names(this, edge, center)
       class(ModelVerticalGrid), intent(inout) :: this
-      character(*), intent(in) :: short_name
+      character(*), optional, intent(in) :: edge
```

Comment:
You need an "unusable" argument here to separate out the options.   Both arguments are the same type (character), so it is dangerously ambiguous which one is intended.

Reply:
Not sure I understand. Both `edge` and `center` are optional arguments. Did you mean something like
```Fortran
   subroutine add_short_names(this, edge, center, unusable)
      class(ModelVerticalGrid), intent(inout) :: this
      character(*), optional, intent(in) :: edge
      class(KeywordEnforder), optional, intent(in) :: unusable
      character(*), optional, intent(in) :: center

      if (present(edge)) this%short_name_edge = edge
      if (present(center)) this%short_name_center = center
      _UNUSED_DUMMY(unusable)
   end subroutine add_short_names
```

Reply:
No  - I mean "unusable" should be even before edge.   Force the user to be explicit since there is no obvious ordering of "center" vs "edge" in this context.

Reply:
Got it. I had seen the `unusable` argument, but was not familiar with its usage. Pushed the change.